metrics-exporters/leadtime-exporter/lt-exporter.py

```diff
@@ -0,0 +1,141 @@
+        self.lead_time = 0
+    def getCommitTime(self):
+        _prefix = "https://api.github.com/repos/"
+        _suffix = "/commits/"
+        myurl = self.repo_url
+        url_tokens = myurl.split("/")
+        url = _prefix + url_tokens[3] + "/" +url_tokens[4].split(".")[0] +_suffix+self.commit_hash
+        response = requests.get(url, auth=(username, token))
+#        print(username, token)
+        commit = response.json()
+#        print(commit)
+        self.commit_time = commit['commit']['committer']['date']
+        self.commit_timestamp = convert_date_time_to_timestamp(self.commit_time)
+    def calculate_lead_time(self):
+        self.lead_time = self.deployment_timestamp - self.commit_timestamp
+
+def match_image_id(replicationController, image_hash):
+    for container in replicationController.spec.template.spec.containers:
+        con_image = container.image
+        image_tokens = con_image.split('@')
+        if len(image_tokens) < 2:
+            return False
+        con_image_hash = image_tokens[1]
+        if con_image_hash == image_hash:
+            return True
+    return False
+
+def convert_date_time_offset_to_timestamp(date_time):
+    date_time = date_time[:-4]
+    timestamp = datetime.strptime(date_time, '%Y-%m-%d %H:%M:%S %z')
+    unixformattime = timestamp.timestamp()
+    return unixformattime
+
+def convert_date_time_to_timestamp(date_time):
+    timestamp = datetime.strptime(date_time, '%Y-%m-%dT%H:%M:%SZ')
+    unixformattime = timestamp.replace(tzinfo=timezone.utc).timestamp()
+    return unixformattime       
+
+def convert_timestamp_to_date_time(timestamp):
+    date_time = datetime(timestamp)
+    return datetime.strftime('%Y-%m-%dT%H:%M:%SZ')
+
+
+def generate_ld_metrics_list(projects):
+
+    urllib3.disable_warnings()
+    k8s_client = config.new_client_from_config()
+    dyn_client = DynamicClient(k8s_client)
+
+    metrics = []
+    if projects == None:
+        v1_projects = dyn_client.resources.get(api_version='project.openshift.io/v1', kind='Project')
+        projects = [ project.metadata.name for project in v1_projects.get().items ]
+
+    for project in projects:    
+        v1_builds = dyn_client.resources.get(api_version='v1',  kind='Build')
+        builds = v1_builds.get(namespace=project)
+
+        v1_replicationControllers = dyn_client.resources.get(api_version='v1',  kind='ReplicationController')
+        replicationControllers = v1_replicationControllers.get(namespace=project)
+
+        print ('Builds Count = ' , len(builds.items))
+        for build in builds.items:
+            if build['spec']['source']['type'] == 'Git':
+                if build['status']['phase'] =='Complete':
+                    app_name = build.metadata.labels.app
+                    metric = mdt_metric(app_name)
+                    metric.build_name=build.metadata.name
+                    metric.build_config_name = build.metadata.labels.buildconfig                    
```

Comment:
I'm wondering if we are fetching/storing more metadata than we need to get the commit hash and timestamp here?  Do we need all these fields to do the timestamp calculation?

Reply:
I am creating a data structure to be used in future exporters if needed, we can strip it down later